provisioner/config.go

```diff
@@ -197,18 +243,127 @@ func (c *VolumeConfig) GetNFServerGraceTime() (int, error) {
 
 // GetFSGroupID fetches the group ID permissions from
 // StorageClass if specified
+// -----------------------------------------------------
+// NOTE: This feature has been deprecated
+//       Alternative: Use FilePermission 'cas.openebs.io/config' annotation
+//                    key on the backend volume PVC. Sample FilePermissions
+//      	      for FSGID-like configuration --
+//
+//                    name: FilePermissions
+//                    data:
+//                      GID: <group-ID>
+//                      mode: "g+s"
+// -----------------------------------------------------
 func (c *VolumeConfig) GetFSGroupID() (*int64, error) {
-	fsGIDStr := c.getValue(FSGroupID)
-	if len(strings.TrimSpace(fsGIDStr)) == 0 {
+	fsGroupIDStr := c.getValue(FSGroupID)
+	if len(strings.TrimSpace(fsGroupIDStr)) == 0 {
 		return nil, nil
 	}
-	fsGIDInt, err := strconv.ParseInt(fsGIDStr, 10, 64)
+	fsGIDInt, err := strconv.ParseInt(fsGroupIDStr, 10, 64)
 	if err != nil {
 		return nil, err
 	}
+
+	klog.Infof("The %s option key '%s' is being deprecated"+
+		" and will be removed in future releases."+
+		"\nYou may use the %s option key in the "+
+		"backend volume PersistentVolumeClaim's %s "+
+		"annotation key to achieve the same result."+
+		"\nSample config:\n"+
+		"\t\t"+"- name: FilePermissions\n"+
+		"\t\t"+"  data:\n"+
+		"\t\t"+"    %s: \"%s\"\n"+
+		"\t\t"+"    %s: \"g+s\"\n",
+		string(mconfig.CASConfigKey), FSGroupID, FilePermissions,
+		string(mconfig.CASConfigKey), FsGID, fsGroupIDStr, FsMode,
+	)
 	return &fsGIDInt, nil
 }
 
+// GetFsGID fetches the group owner's ID from
+// PVC annotation, if specified
+func (c *VolumeConfig) GetFsGID() (string, error) {
+	fsGIDStr := strings.TrimSpace(c.getData(FilePermissions, FsGID))
+
+	// TODO: remove this block when FSGID is deprecated
+	// Replace this block with the commented code below
+	fsGroupIDStr, _ := c.GetFSGroupID()
+
+	existsFsGIDStr := (len(fsGIDStr) > 0)
+	existsFsGroupIDStr := (fsGroupIDStr != nil)
+
+	// Checking if FSGID and FilePermissions (GID) are being used together
+	if existsFsGIDStr && existsFsGroupIDStr {
+		return "", errors.Errorf("both '%s' and %s data "+
+			"key '%s' cannot be used together",
+			FSGroupID, FilePermissions, FsGID,
+		)
+	}
+
+	if existsFsGroupIDStr {
+		return "", nil
+	} else if existsFsGIDStr {
+		return fsGIDStr, nil
+	}
+
+	return "", nil
+
+	/*
+		if len(fsGIDStr) == 0 {
+			return ""
+		}
+
+		return fsGIDStr
+	*/
```

Comment:
nit: Below should work... similar in GetFsMode func also
```suggestion
	}
	return fsGIDStr
```

Reply:
Included this suggested change in GetFsMode() and GetFsGID() methods.